crmd/lrm_state.c

```diff
@@ -580,122 +516,37 @@ remote_proxy_cb(lrmd_t *lrmd, void *userdata, xmlNode *msg)
 
         crm_warn("Reconnection attempts to %s may result in failures that must be cleared",
                  lrm_state->node_name);
-        return;
 
-    } else if (safe_str_eq(op, LRMD_IPC_OP_NEW)) {
-        int rc;
-        const char *channel = crm_element_value(msg, F_LRMD_IPC_IPC_SERVER);
+    } else if (safe_str_eq(op, LRMD_IPC_OP_REQUEST) && proxy->is_local) {
+        /* this is for the crmd, which we are, so don't try
+         * and connect/send to ourselves over ipc. instead
+         * do it directly.
+         */
+        int flags = 0;
+        xmlNode *request = get_message_xml(msg, F_LRMD_IPC_MSG);
```

Comment:
Previously for this case, it would also do:

        proxy->last_request_id = 0;
        crm_xml_add(request, XML_ACL_TAG_ROLE, "pacemaker-remote");
    #if ENABLE_ACL
        CRM_ASSERT(lrm_state->node_name);
        crm_acl_get_set_user(request, F_LRMD_IPC_USER, lrm_state->node_name);
    #endif

Is that not needed?